In [1]:
#Import library for data visualization
import pandas as pd 

In [2]:
#Load the dataset into DataFrame
df = pd.read_csv("KaggleV2-May-2016.csv")

In [3]:
#Display the first five rows 
df.head()

,PatientId,AppointmentID,Gender,ScheduledDay,AppointmentDay,Age,Neighbourhood,Scholarship,Hipertension,Diabetes,Alcoholism,Handcap,SMS_received,No-show
0,2.987250e+13,5642903,F,2016-04-29T18:38:08Z,2016-04-29T00:00:00Z,62,JARDIM DA PENHA,0,1,0,0,0,0,No
1,5.589978e+14,5642503,M,2016-04-29T16:08:27Z,2016-04-29T00:00:00Z,56,JARDIM DA PENHA,0,0,0,0,0,0,No
2,4.262962e+12,5642549,F,2016-04-29T16:19:04Z,2016-04-29T00:00:00Z,62,MATA DA PRAIA,0,0,0,0,0,0,No
3,8.679512e+11,5642828,F,2016-04-29T17:29:31Z,2016-04-29T00:00:00Z,8,PONTAL DE CAMBURI,0,0,0,0,0,0,No
4,8.841186e+12,5642494,F,2016-04-29T16:07:23Z,2016-04-29T00:00:00Z,56,JARDIM DA PENHA,0,1,1,0,0,0,No


In [16]:
#Checking the rows and columns the dataset 
print("Shape:", df.shape)

Shape: (110527, 14)


In [11]:
#View column names, data types, and non-null counts
print("Dataset Info")
df.info()

Dataset Info
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110527 entries, 0 to 110526
Data columns (total 14 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   PatientId       110527 non-null  float64
 1   AppointmentID   110527 non-null  int64  
 2   Gender          110527 non-null  object 
 3   ScheduledDay    110527 non-null  object 
 4   AppointmentDay  110527 non-null  object 
 5   Age             110527 non-null  int64  
 6   Neighbourhood   110527 non-null  object 
 7   Scholarship     110527 non-null  int64  
 8   Hipertension    110527 non-null  int64  
 9   Diabetes        110527 non-null  int64  
 10  Alcoholism      110527 non-null  int64  
 11  Handcap         110527 non-null  int64  
 12  SMS_received    110527 non-null  int64  
 13  No-show         110527 non-null  object 
dtypes: float64(1), int64(8), object(5)
memory usage: 11.8+ MB


In [10]:
#Check the missing values in the datset
print("Missing Values")
print(df.isnull().sum())

Missing Values
PatientId         0
AppointmentID     0
Gender            0
ScheduledDay      0
AppointmentDay    0
Age               0
Neighbourhood     0
Scholarship       0
Hipertension      0
Diabetes          0
Alcoholism        0
Handcap           0
SMS_received      0
No-show           0
dtype: int64


In [17]:
# Check if there are any duplicates in the datset
print("Duplicates before cleaning:", df.duplicated().sum())

Duplicates before cleaning: 0


In [22]:
#Check gender values before standarization
print("Before standarization:")
print(df["Gender"].value_counts())

#Remove spaces and convert uppercase
df["Gender"] = df["Gender"].str.strip().str.upper()

#Confirmation
print("\n After standarization:")
print(df["Gender"].value_counts())

Before standarization:
Gender
F    71840
M    38687
Name: count, dtype: int64

 After standarization:
Gender
F    71840
M    38687
Name: count, dtype: int64


In [32]:
#View the date colums before conversion
print(df["ScheduledDay"].head())
print()
print(df["AppointmentDay"].head())

#Convert date columns to proper date format
df["ScheduledDay"]= pd.to_datetime(df["ScheduledDay"])
df["AppointmentDay"]= pd.to_datetime(df["AppointmentDay"])
print()
#Connfirm the date colums 
print(df[["ScheduledDay", "AppointmentDay"]]. dtypes)

0   2016-04-29 18:38:08+00:00
1   2016-04-29 16:08:27+00:00
2   2016-04-29 16:19:04+00:00
3   2016-04-29 17:29:31+00:00
4   2016-04-29 16:07:23+00:00
Name: ScheduledDay, dtype: datetime64[ns, UTC]

0   2016-04-29 00:00:00+00:00
1   2016-04-29 00:00:00+00:00
2   2016-04-29 00:00:00+00:00
3   2016-04-29 00:00:00+00:00
4   2016-04-29 00:00:00+00:00
Name: AppointmentDay, dtype: datetime64[ns, UTC]

ScheduledDay      datetime64[ns, UTC]
AppointmentDay    datetime64[ns, UTC]
dtype: object


In [42]:
#Remove rows where age is less than 0
before=df.shape[0]
df=df[df["Age"] >= 0]
after = df.shape[0]

#I wrote one 1 because i mistakenly cleaned the data before recording my answers
print("Rows removed:", 1)
print("Rows remaining:", after)

Rows removed: 1
Rows remaining: 110526


In [43]:
#View first five rows of cleaned datset
df.head()

,PatientId,AppointmentID,Gender,ScheduledDay,AppointmentDay,Age,Neighbourhood,Scholarship,Hipertension,Diabetes,Alcoholism,Handcap,SMS_received,No-show
0,2.987250e+13,5642903,F,2016-04-29 18:38:08+00:00,2016-04-29 00:00:00+00:00,62,JARDIM DA PENHA,0,1,0,0,0,0,No
1,5.589978e+14,5642503,M,2016-04-29 16:08:27+00:00,2016-04-29 00:00:00+00:00,56,JARDIM DA PENHA,0,0,0,0,0,0,No
2,4.262962e+12,5642549,F,2016-04-29 16:19:04+00:00,2016-04-29 00:00:00+00:00,62,MATA DA PRAIA,0,0,0,0,0,0,No
3,8.679512e+11,5642828,F,2016-04-29 17:29:31+00:00,2016-04-29 00:00:00+00:00,8,PONTAL DE CAMBURI,0,0,0,0,0,0,No
4,8.841186e+12,5642494,F,2016-04-29 16:07:23+00:00,2016-04-29 00:00:00+00:00,56,JARDIM DA PENHA,0,1,1,0,0,0,No


In [44]:
#Confirm final data type and non-null counts 
df.info

<bound method DataFrame.info of            PatientId  AppointmentID Gender              ScheduledDay  \
0       2.987250e+13        5642903      F 2016-04-29 18:38:08+00:00   
1       5.589978e+14        5642503      M 2016-04-29 16:08:27+00:00   
2       4.262962e+12        5642549      F 2016-04-29 16:19:04+00:00   
3       8.679512e+11        5642828      F 2016-04-29 17:29:31+00:00   
4       8.841186e+12        5642494      F 2016-04-29 16:07:23+00:00   
...              ...            ...    ...                       ...   
110522  2.572134e+12        5651768      F 2016-05-03 09:15:35+00:00   
110523  3.596266e+12        5650093      F 2016-05-03 07:27:33+00:00   
110524  1.557663e+13        5630692      F 2016-04-27 16:03:52+00:00   
110525  9.213493e+13        5630323      F 2016-04-27 15:09:23+00:00   
110526  3.775115e+14        5629448      F 2016-04-27 13:30:56+00:00   

                  AppointmentDay  Age      Neighbourhood  Scholarship  \
0      2016-04-29 00:00:00+00:

In [46]:
#Saving the cleaned datset to a new csv file 
df. to_csv("medical_appointments_cleaned.csv", index=False)

##Summary of Changes Made 
1. Loaded dataset- 110,527 rows and columns
2. Checked missing values- No missing values found
3. Checked duplicates- No duplicates rows found
4. Cleaned column names- No changes needed
5. Standardized Gender Column- Stripped spaces and converted to upper case(F,M)
6. Converted date columns- ScheduledDay and AppointmentDay converted from object to datetime format
7. Removed invalid ages- 1 row with invalid age(-1) removed